evaluate_performance.py

```diff
@@ -0,0 +1,130 @@
+import json
+
+from functools import partial
+from sklearn.metrics import (
+    accuracy_score,
+    precision_score,
+    recall_score,
+    f1_score,
+    confusion_matrix,
+)
+from run_llm.benchmark_set_up import Benchmark, get_benchmark_by_name
+
+# Configuration
+# =============
+PATH = "results.json"  # Path to the results file
+PRINT_ALL = False  # Print all groups or just the top 5
+INCLUDE_ALL_METRICS_IN_PROPERTY = False  # Include all metrics in the property ranking
+RANK_BY = "f1"  # This is due to the unbalanced data. Can be: "accuracy", "precision", "recall", "f1"
+MIN_SAMPLES = 3  # Minimum number of samples to consider a group
+SAVE_RESULT_PATH = (
+    None  # If saving to file set it to something like: "eval_results.txt"
+)
+OUTPUT_FILE = open(SAVE_RESULT_PATH, "w") if SAVE_RESULT_PATH else None
+
+printo = partial(print, file=OUTPUT_FILE)
+
+
+# Functions
+# =========
+def calculate_metrics(ground_truths: list[str], predictions: list[str]):
+    return {
+        "accuracy": accuracy_score(ground_truths, predictions),
+        "precision": precision_score(
+            ground_truths, predictions, average="macro", zero_division=0
```

Comment:
Using `average="macro"` will calculate the unweighted mean, which would not be desirable for unbalanced datasets. Would it not be preferable to return the score for each class?

Reply:
We can return the "weighted", all of this can be changed under the configuration easily. The issue with score is that it returns a list, which cannot be ranked for display. I'll set the default as score for now and then for the ranking I just do a naive mean of the score list.